src/services/test-helpers/mock/mockApi.ts

```diff
@@ -517,6 +519,159 @@ const auctionsWinningsAt = () =>
+			approvals: rococoRegistry.createType('u32', 20),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetDetails', responseObj);
+	});
+
+const assetsMetadata = () =>
+	Promise.resolve().then(() => {
+		const responseObj = {
+			deposit: balanceOfTwo,
+			name: rococoRegistry.createType('Bytes', 'statemint'),
+			symbol: rococoRegistry.createType('Bytes', 'DOT'),
+			decimals: rococoRegistry.createType('u8', 10),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetMetadata', responseObj);
+	});
+
+const assetBalanceObjOne = {
+	balance: assetTBalanceOne,
+	isFrozen: falseBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjTwo = {
+	balance: assetTBalanceTwo,
+	isFrozen: trueBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjThree = {
+	balance: assetTBalanceTwo,
+	isFrozen: falseBool,
+	isSufficient: falseBool,
+};
+
+const assetBalanceFactory = {
+	'10': rococoRegistry.createType('AssetBalance', assetBalanceObjOne),
+	'20': rococoRegistry.createType('AssetBalance', assetBalanceObjTwo),
+	'30': rococoRegistry.createType('AssetBalance', assetBalanceObjThree),
+};
+
+const assetAccountStorageKeyOne = statemintTypeFactory.storageKey(
+	10,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyTwo = statemintTypeFactory.storageKey(
+	20,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyThree = statemintTypeFactory.storageKey(
+	30,
+	'AssetId',
+	statemintApiV1.query.assets.account,
+	accountIdOne
+);
+
+const assetsAccountKeysAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			assetAccountStorageKeyOne,
+			assetAccountStorageKeyTwo,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+		];
+	});
+
+/**
+ * @param assetId options are 10, 20, 30
+ */
+interface AssetsAccount {
+	// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+	[key: string]: any;
+}
+
+const assetsAccount: AssetsAccount = (
+	assetId: number | AssetId,
+	_address: string
+) => {
+	const id = typeof assetId === 'number' ? assetId : assetId.toNumber();
+
+	return Promise.resolve().then(() => {
+		switch (id) {
+			case 10:
+				return assetBalanceFactory[10];
+			case 20:
+				return assetBalanceFactory[20];
+			case 30:
+				return assetBalanceFactory[30];
+			default:
+				return;
+		}
+	});
+};
+
+assetsAccount.extend = function (args: AssetsAccount) {
+	for (const i in args) {
+		// eslint-disable-next-line  @typescript-eslint/no-unsafe-assignment
+		this[i] = args[i];
+	}
+	return this;
+};
+
+// eslint-disable-next-line  @typescript-eslint/no-unsafe-call
+assetsAccount.extend({
+	keysAt: assetsAccountKeysAt,
+});
+
+const assetApprovals = () =>
+	Promise.resolve().then(() => {
+		const assetObj = {
+			amount: assetTBalanceOne,
+			deposit: balanceOfTwo,
+		};
+		return rococoRegistry.createType('Option<AssetApproval>', assetObj);
+	});
```

Comment:
```suggestion
const assetApprovals = async () => {
	const assetObj = {
		amount: assetTBalanceOne,
		deposit: balanceOfTwo,
	};
	return rococoRegistry.createType('Option<AssetApproval>', assetObj);
};
```

Does this need to return a promise?

Reply:
Yeah, since its mocking a response to a storage query the code expects a `Promise`

Reply:
No it technically doesn't, but I do it here to keep the format the same as the rest of the functions affiliated with the mockApi in the file.